client/src/components/TreeDropdownField/TreeDropdownField.js

```diff
@@ -116,7 +114,7 @@ class TreeDropdownField extends Component {
    * @return {Object}
    */
   getVisibleTree() {
-    return this.findTreeByPath(this.props.tree, this.props.visible);
+    return this.props.findTreeByPath(this.props.tree, this.props.visible);
```

Comment:
It's a bit of a halfway measure, mostly owing to the limitations of Injector. The idea is to give the user some say in how the visible tree is computed and rendered. (https://github.com/silverstripe/silverstripe-asset-admin/pull/592/files#diff-5ec27365abb27c9225a02be39af02f85R23)

It abstracts away how the tree is parsed, but assigns the utilities as default props so that it doesn't impose undue composition on the developer.